server.go

```diff
@@ -116,8 +129,52 @@ func (s *server) Start() {
 	s.fundingMgr.Start()
 	s.htlcSwitch.Start()
 
+	// ROUTING ADDED
+	s.routingMgr.Start()
+
 	s.wg.Add(1)
 	go s.queryHandler()
+
+	// ROUTING ADDED
+	s.wg.Add(1)
+	go s.routingOutHandler()
+}
+
+// ROUTING ADDED
+// Handles outgoing routing messages
+// Determines correct peer and forward message to it
+func (s *server) routingOutHandler(){
```

Comment:
A distinct goroutine for forwarding messages from the routing manager to peers seems a bit unnecessary. Additionally, the operation of this goroutine might cause a race condition around access to the `peers` map with the current `queryHandler` goroutine. 

I think we should instead roll this into the current `queryHandler` goroutine. With some slight refactoring we can extract of the logic for searching for a particular peer by their `lightningID`. Actually, we may just want to introduce an additional index (map) for quick-lookup since this might be performed rather frequently, and will result in more succinct logic.